IPython/core/profiledir.py

```diff
@@ -109,16 +124,20 @@ def check_log_dir(self, change=None):
     
     @observe('startup_dir')
     def check_startup_dir(self, change=None):
-        self._mkdir(self.startup_dir)
-
-        readme = os.path.join(self.startup_dir, 'README')
-        src = os.path.join(get_ipython_package_dir(), u'core', u'profile', u'README_STARTUP')
-
-        if not os.path.exists(src):
-            self.log.warning("Could not copy README_STARTUP to startup dir. Source file %s does not exist.", src)
-
-        if os.path.exists(src) and not os.path.exists(readme):
-            shutil.copy(src, readme)
+        if self._mkdir(self.startup_dir):
+            readme = os.path.join(self.startup_dir, "README")
+            src = os.path.join(
+                get_ipython_package_dir(), "core", "profile", "README_STARTUP"
+            )
+
+            if not os.path.exists(src):
+                self.log.warning(
+                    "Could not copy README_STARTUP to startup dir. Source file %s does not exist.",
+                    src,
+                )
+
+            if os.path.exists(src) and not os.path.exists(readme):
+                shutil.copy(src, readme)
```

Comment:
Small suggestion to avoid multiple `os.path.exists()` calls.

```suggestion
            if os.path.exists(src):
                if not os.path.exists(readme):
                    shutil.copy(src, readme)
            else:
                self.log.warning(
                    "Could not copy README_STARTUP to startup dir. Source file %s does not exist.",
                    src,
                )
```